### ISSUE TITLE
Can't create Jupyter pods due to length of username

### ISSUE THREAD

**Issue by durandom** [opendatahub-io/opendatahub-operator#490]
Usernames can be longer than 42 characters, e.g. mapping username from Active Directory or mapping the email address.
In this case the constructed name for the pod in the Jupyter stateful set exceeds the 63 character limit of kubernetes for labels.

```
statefulset-controller create Pod jupyter-nb-Luhwvfk2nyoLopbjs71aw-2dsxSenuawajyrftbyhktba-0 in StatefulSetjupyter-nb-luhwvfk2nyolopbjs71aw-2dsxSenuawaiyrftbyhktba failed error: Pod "jupyte
r-nb-luhwvfk2nyolopbis7iaw-2ds×5enuawaiyrftbyhktba-o" is invalid: metadata. labels: Invalid value:
"jupyter-nb-Luhwvfk2nyoLopb]s7iaw-2d5x5enuawaiyrftbyhktba-f9ccb69cg"; must be no more than 63 characters
```

Is there any workaround currently available, other than making sure that the username does not exceed 42 characters?

**Comment by zdtsw** [comment]
isnt that something as the label `controller-revision-hash` created on statefulset pod automatically by k8s, which append extra 10 characters? 
plus the suffix `"jupyter-nb-"` with 10 characters,  and the limitation of k8s that label with max 64 characters, 
so the user name can only limit to 63-10-10 = 43

**Comment by durandom** [comment]
Correct. I think this limit is too short, eg if you have the UID of active directory or an email as the username. 
Would be better if the username gets hashed to avoid that limit

**Comment by zdtsw** [comment]
FYI @harshad16

**Comment by harshad16** [comment]
Thanks for pointing me to this.
I would have to take a look at our kubeflow-notebook controller level, to get this fixed.

**Comment by zdtsw** [comment]
should we transfer this issue to https://github.com/opendatahub-io/kubeflow?
